lib/core_extensions.rb

```diff
@@ -38,8 +38,12 @@ def initialize(*attribute)
     end
 
     def before_destroy(record)
-      klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
+      klasses.each do |klass, foreign_key|
+        record_class = record.class
+        association = record_class.reflect_on_association(klass)
+        foreign_key ||= "#{record_class.name}_id".to_sym
+        related_associations = association.class_name.constantize.unscoped.where(foreign_key => record.id)
```

Comment:
`Host.unscoped.where(:subnet_id => 123)` won't work since subnet is obtained through primary_interface.
I can change this to Host.send(:subnet) for each host and compare to the subnet but that will take a long time when there are many hosts.